app/src/main/java/org/oppia/android/app/options/AppLanguageFragmentPresenter.kt

```diff
@@ -59,8 +74,37 @@ class AppLanguageFragmentPresenter @Inject constructor(
     }
   }
 
-  fun onLanguageSelected(selectedLanguage: String) {
+  fun onLanguageSelected(selectedLanguage: OppiaLanguage) {
     appLanguageSelectionViewModel.selectedLanguage.value = selectedLanguage
     updateAppLanguage(selectedLanguage)
+    updateAppLanguageSelection(selectedLanguage)
+  }
+
+  private fun updateAppLanguageSelection(oppiaLanguage: OppiaLanguage) {
+    val appLanguageSelection = AppLanguageSelection.newBuilder().apply {
+      selectedLanguage = oppiaLanguage
+      selectedLanguageValue = oppiaLanguage.number
+    }.build()
+
+    val userProfileId = ProfileId.newBuilder().apply { internalId = profileId!! }.build()
+
+    if (userProfileId != null) {
```

Comment:
This can't be null since it's being built right before this.

Reply:
Updated.